src/databricks/labs/blueprint/paths.py

```diff
@@ -520,7 +520,18 @@ def resolve(self: P, strict: bool = False) -> P:
         if strict and not absolute.exists():
             msg = f"Path does not exist: {self}"
             raise FileNotFoundError(msg)
-        return absolute
+        return absolute.normalize()
+
+    def normalize(self: P) -> P:
+        for index, part in enumerate(self._path_parts):
+            # we can't normalize '/../stuff' so let's ignore such scenarios
+            if index == 0 or part != "..":
+                continue
+            segments = list(self._path_parts)
+            segments.pop(index)
+            segments.pop(index - 1)
+            return self.with_segments(self.anchor, *segments).normalize()
+        return self
```

Comment:
```suggestion
        stack = []
        for part in self._path_parts:
            if part == "..":
                if stack:
                    stack.pop()
            elif part == "." or not part:
                continue
            else:
                stack.append(part)
        return self.with_segments(self.anchor, *stack)
```

can we perhaps get a non-recursive option?

Reply:
done